File: components/credentials/src/main/java/org/cloudfoundry/credhub/requests/PasswordGenerateRequest.java
package org.cloudfoundry.credhub.requests;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

public class PasswordGenerateRequest extends BaseCredentialGenerateRequest {
  @JsonProperty("parameters")
  private StringGenerationParameters generationParameters;

  @Override
  @JsonIgnore
  public GenerationParameters getGenerationParameters() {
    if (generationParameters == null) {
      generationParameters = new StringGenerationParameters();
    }
    return generationParameters;
  }

  public void setGenerationParameters(final StringGenerationParameters generationParameters) {
    this.generationParameters = generationParameters;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/PasswordGenerateRequest.kt
package org.cloudfoundry.credhub.requests

import com.fasterxml.jackson.annotation.JsonProperty

class PasswordGenerateRequest : BaseCredentialGenerateRequest() {
    @JsonProperty("parameters")
    override var generationParameters: StringGenerationParameters? = null
        get() {
            if (field == null) {
                field = StringGenerationParameters()
            }
            return field
        }
}
